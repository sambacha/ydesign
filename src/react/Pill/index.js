import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Text from '../Text';

const StyledPill = styled(Text)`
  & {
    position: relative;
    overflow: hidden;
  }
  &::after {
    content: '';
    display: block;
    position: absolute;
    left: 0;
    right: 0;
    top: 0;
    height: 100%;
    width: 100%;
    opacity: 0.12;
    background: currentColor;
    pointer-events: none;
  }
`;

const Pill = React.forwardRef((props, ref) => (
  <StyledPill ref={ref} {...props} />
));

Pill.defaultProps = {
  color: 'dark-gray',
  bg: '#FFF',
  fontSize: 1,
  fontWeight: 2,
  lineHeight: 'solid',
  height: '2rem',
  px: 3,
  py: 1,
  borderRadius: 3,
  display: 'inline-flex',
  alignItems: 'center',
};

Pill.propTypes = {
  ...Text.propTypes,
};

Pill.displayName = 'Pill';

export default Pill;
