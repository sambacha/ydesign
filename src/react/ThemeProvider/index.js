import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
