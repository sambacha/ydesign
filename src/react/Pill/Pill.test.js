import Pill from './';
import ThemeProvider from '../ThemeProvider';
import theme from '../theme';

describe('Pill component sanity', () => {
  it('has name', () => {
    expect(Pill.displayName).toBe('Pill');
  });

  it('matches default snapshot', () => {
    const component = render(<Pill color={'blue'} />);
    expect(component).toMatchSnapshot();
  });

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Pill color={'blue'} />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
