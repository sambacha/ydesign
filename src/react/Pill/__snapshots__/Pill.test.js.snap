// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #3259D6;
  background-color: #FFF;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: relative;
  overflow: hidden;
}

.c1::after {
  content: '';
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  height: 100%;
  width: 100%;
  opacity: 0.12;
  background: currentColor;
  pointer-events: none;
}

<div
  class="c0 c1"
  color="blue"
  display="inline-flex"
  font-size="1"
  font-weight="2"
  height="2rem"
/>
`;

exports[`Pill component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  color: #3259D6;
  background-color: #FFF;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 4px;
  padding-bottom: 4px;
  border-radius: 16px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: relative;
  overflow: hidden;
}

.c1::after {
  content: '';
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  height: 100%;
  width: 100%;
  opacity: 0.12;
  background: currentColor;
  pointer-events: none;
}

<div
  class="c0 c1"
  color="blue"
  display="inline-flex"
  font-size="1"
  font-weight="2"
  height="2rem"
/>
`;
